R/knitr-engine.R

```diff
@@ -294,6 +294,7 @@ eng_python <- function(options) {
 }
 
 eng_python_initialize <- function(options, envir) {
+  if (missing(envir)) envir <- environment()
```

Comment:
nit: I'd prefer we be explicit about which environment is used here

Reply:
Should it be a default value? This function was called in a single place with the argument `envir = environment()` (which looks a lot as a default value). I needed it to be called in a second place, and may need it in a third one in the future.

I don't know why I didn't change it to be a default value right away. Maybe I was thinking with a Python mind, for which dynamic values can't be default argument values...

By the way, are there any implications of calling `eng_python_initialize()` multiple times for a single `knitr` chunk? To the extent that I understood the underlying code called by it, there seems to be no problem, neither of side effects, nor of significant overhead.

Reply:
Sorry, I mistook `environment()` for `new.env()`. So it made sense to call `environment()` inside the `eng_python()` call frame. However, I tracked down where this environment is used and discovered that **it's used nowhere**! It's ultimately passed to `eng_python_initialize_matplotlib()` and `eng_python_initialize_plotly()`, which don't use it at all. Any reason for keeping this parameter?

Reply:
I think we can remove it.